docs/data/tree-view/rich-tree-view/headless/headless.md

```diff
@@ -0,0 +1,279 @@
+const useOtherCustomPlugin = ({ models, instance }) => {
+  const handleSomeAction = () => {
+    instance.toggleCustomModel();
+  };
+};
+```
+
+### Emit and receive events
+
+```ts
+const useCustomPlugin = () => {
+  const toggleCustomModel = () => {
+    const newValue = !models.customModel.value;
+    models.customModel.setValue(newValue);
+    publishTreeViewEvent(instance, 'toggleCustomModel', { value: newValue });
+  };
+
+  populateInstance(instance, { toggleCustomModel });
+};
+```
+
+You can then subscribe to this event in any other plugin:
+
+```ts
+const useOtherCustomPlugin = ({ instance }) => {
+  useInstanceEventHandler(instance, 'toggleCustomModel', ({ value }) => {
+    console.log('New value of customModel', value);
+  });
+};
+```
+
+:::info
+If you are using TypeScript, you need to define your dependencies in your plugin signature, see the [Plugin typing](/x/react-tree-view/rich-tree-view/headless/#plugin-typing) section for more information.
+:::
+
+### Pass props to your root element
+
+You can use the `getRootProps` property of your returned value to pass props to your root element:
+
+```ts
+const useCustomPlugin = ({ params }) => {
+  return {
+    getRootProps: () => ({
+      'data-customparam': params.customParam,
+    }),
+  };
+};
+```
+
+### Plugin typing
+
+The typing of a plugin is defined using its _signature_.
+This type contains the following information:
+
+```ts
+type UseCustomPluginSignature = TreeViewPluginSignature<
+  // The params specific to your plugin before running `getDefaultizedParams`
+  UseCustomPluginParams,
+  // The params specific to your plugins after running `getDefaultizedParams`
+  UseCustomPluginDefaultizedParams,
+  // The methods added to the tree view instance by your plugin
+  UseCustomPluginInstance,
+  // The events emitted by your plugin
+  UseCustomPluginEvents,
+  // The states defined by your plugin
+  UseCustomPluginState,
+  // The name of the models defined by your plugin
+  UseCustomPluginModelNames,
+  // The plugins this plugin needs to work correctly
+  UseCustomPluginDependantPlugins
+>;
+```
+
+The most basic plugin would have the following signature:
+
+```ts
+type UseCustomPluginSignature = TreeViewPluginSignature<
+  {},
+  {},
+  {},
+  {},
+  {},
+  never,
+  []
+>;
+```
+
+The plugin built in the sections above would have the following signature:
+
+```ts
+type UseCustomPluginSignature = TreeViewPluginSignature<
+  // Params
+  { customParam?: boolean; customModel?: boolean; defaultCustomModel?: boolean },
+  // Defaultized params
+  // `customParam` and `defaultCustomModel` have a default value defined in `getDefaultizedParams`
+  { customParam: boolean; customModel?: boolean; defaultCustomModel: boolean },
+  // Instance
+  { toggleCustomModel: () => void },
+  // Events
+  'toggleCustomModel',
+  // State
+  {},
+  // Model names
+  'customModel',
+  // Dependant plugns,
+  [UseTreeViewExpansionSignature]
+>;
+```
+
+## Examples
+
+### Log expanded items
```

Comment:
I just copied the existing example, but we should clearly improve it before making this page public.

IMHO this is not the focus of this PR.